src/Symfony/Bridge/Twig/NodeVisitor/TranslationDefaultDomainNodeVisitor.php

```diff
@@ -49,19 +47,25 @@ public function enterNode(Node $node, Environment $env): Node
         }
 
         if ($node instanceof TransDefaultDomainNode) {
+            ++$this->nestingLevel;
+
             if ($node->getNode('expr') instanceof ConstantExpression) {
                 $this->scope->set('domain', $node->getNode('expr'));
 
                 return $node;
             }
 
-            $name = class_exists(AssignContextVariable::class) ? new AssignContextVariable(self::INTERNAL_VAR_NAME, $node->getTemplateLine()) : new AssignNameExpression(self::INTERNAL_VAR_NAME, $node->getTemplateLine());
-            $this->scope->set('domain', class_exists(ContextVariable::class) ? new ContextVariable(self::INTERNAL_VAR_NAME, $node->getTemplateLine()) : new NameExpression(self::INTERNAL_VAR_NAME, $node->getTemplateLine()));
-
             if (class_exists(Nodes::class)) {
+                $name = new LocalVariable(null, $node->getTemplateLine());
```

Comment:
@xabbuh changing this to a local variable is likely the root cause of https://github.com/EasyCorp/EasyAdminBundle/issues/6605 because a local variable will not be available in blocks (it won't stay available in the Twig child scopes that are totally separate PHP scopes in the compiled template).

To me, we should keep assigning a context variable, as done before your PR.